### PR TITLE
MES-89 - Refactoring and Adding the ability to set an Activity Code

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

- Updated Test Schema to 1.0.12
- Refactored the tests reducer 
- Added ability to set activity code
- Refactored Calculate Test Result to use new Activity Code Action and removing the local state result storage
- Added Enum for Activity Codes

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/modules/tests/tests.reducer.ts [hunk 1]
```diff
@@ -43,26 +32,42 @@ const initialState: TestsModel = {
...
+  return (state.currentTest && state.currentTest.slotId) ? state.currentTest.slotId : null;
+};
 
+const createStateObject = (state: TestsModel, action: Action, slotId: string) => {
   return {
     ...state,
     startedTests: {
       ...state.startedTests,
       [slotId]: {
-        // Each sub-reducer deals with state scoped to a specific test and has no knowledge of
```
Comment (ammarhaiderbjss) [inline src/modules/tests/tests.reducer.ts]: Can we keep this comment in? It's useful for people new to the codebase

Comment (Cavalier07) [reply src/modules/tests/tests.reducer.ts]: done :)

File: src/modules/tests/tests.reducer.ts [hunk 2]
```diff
@@ -43,17 +32,37 @@ const initialState: TestsModel = {
...
+          },
+        },
+      };
+    default:
+      return slotId ? createStateObject(state, action, slotId) : state;
+  }
+}
+
+const deriveSlotId = (state: TestsModel, action: Action): string | null => {
+  if (action instanceof journalActions.StartTest) {
```
Comment (johnmorleybjss) [inline src/modules/tests/tests.reducer.ts]: Nice refactor of this, needs a rebase to pick up the additional action `journalActions.ActivateTest` that will resolve a slot ID.

File: src/shared/models/activity-codes.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  TRAFFIC = '67',
+  NATURAL_DISASTER = '68',
+  LICENCE_FAILED_CHECK = '69',
+  CANDIDATE_REFUSED_TO_SIGN_RESIDENCY_DECLARATION = '70',
+  CANDIDATE_STOPS_TEST = '71',
+  ILLEGAL_ACTIVITY_FROM_CANDIDATE = '73',
+  INDUSTRIAL_ACTION = '74',
+  AUTHORISED_OCCUPANT_INTERVENED_DURING_TEST = '75',
+  INCORRECT_LENSE_WORN = '82',
+  SITE_ACCESS_MANAGER_NOT_AVAILBLE = '83',
```
Comment (rossfauldsbjss) [inline src/shared/models/activity-codes.ts]: AVAILBLE should be AVAILABLE

File: src/modules/tests/tests.reducer.ts (hunk 2 above)
Comment (Cavalier07) [reply src/modules/tests/tests.reducer.ts]: Just tried to rebase and there is nothing to add to this pr as I'm on the latest code

Comment (johnmorleybjss) [reply src/modules/tests/tests.reducer.ts]: I'm referring to this line:
https://github.com/dvsa/mes-mobile-app/blob/develop/src/modules/tests/tests.reducer.ts#L98

File: src/shared/models/activity-codes.ts (hunk 3 above)
Comment (Cavalier07) [reply src/shared/models/activity-codes.ts]: Fixed :)

File: src/modules/tests/tests.reducer.ts (hunk 2 above)
Comment (Cavalier07) [reply src/modules/tests/tests.reducer.ts]: Thanks had missed that in the rebase, just pushing up a fix for it :)

**Pull request merged**